homeassistant/components/integration/sensor.py

```diff
@@ -196,12 +196,13 @@ def calc_integration(event):
             # or device_class.
             update_state = False
 
-            unit = new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-            if unit is not None:
-                new_unit_of_measurement = self._unit_template.format(unit)
-                if self._unit_of_measurement != new_unit_of_measurement:
-                    self._unit_of_measurement = new_unit_of_measurement
-                    update_state = True
+            if new_state.attributes:
```

Comment:
This will not cover the issue in #71967 

`AttributeError: 'NoneType' object has no attribute 'attributes'`

Something like:
```
            if new_state and ATTR_UNIT_OF_MEASUREMENT in new_state.attributes:
```

Reply:
ah rats, you're right. I was under the impression that the lack of 'attributes' being present in the state was causing this issue.

In that case I think it's best to pull  the None / Unknown handling of new_state that's already in there to the top, since anything else before the current location of that early return does depend on it.